plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/CloudSdkServerLaunchTab.java

```diff
@@ -0,0 +1,157 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package com.google.cloud.tools.eclipse.appengine.localserver.ui;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.ui.AbstractLaunchConfigurationTab;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.TraverseEvent;
+import org.eclipse.swt.events.TraverseListener;
+import org.eclipse.swt.graphics.Font;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Text;
+
+import com.google.cloud.tools.eclipse.appengine.localserver.Activator;
+import com.google.cloud.tools.eclipse.appengine.localserver.server.CloudSdkServer;
+
+/**
+ * A launch configuration tab that displays and edits "gcloud app preview"
+ * command flags for the Cloud SDK server.
+ */
+public class CloudSdkServerLaunchTab extends AbstractLaunchConfigurationTab {
+  protected Text programFlagText;
+
+  @Override
+  public void createControl(Composite parent) {
+    Font font = parent.getFont();
+    Composite comp = new Composite(parent, SWT.NONE);
+    GridLayout layout = new GridLayout(1, true);
+    comp.setLayout(layout);
+    comp.setFont(font);
+
+    GridData gd = new GridData(GridData.FILL_BOTH);
+    comp.setLayoutData(gd);
+    setControl(comp);
+
+    Group group = new Group(comp, SWT.NONE);
+    group.setFont(font);
+    layout = new GridLayout();
+    group.setLayout(layout);
+    group.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+    String controlName = ("Program &flags");
+    group.setText(controlName);
+
+    programFlagText = new Text(group, SWT.MULTI | SWT.WRAP | SWT.BORDER | SWT.V_SCROLL);
+    programFlagText.addTraverseListener(new TraverseListener() {
+      public void keyTraversed(TraverseEvent e) {
+        switch (e.detail) {
+        case SWT.TRAVERSE_ESCAPE:
+        case SWT.TRAVERSE_PAGE_NEXT:
+        case SWT.TRAVERSE_PAGE_PREVIOUS:
+          e.doit = true;
+          break;
+        case SWT.TRAVERSE_RETURN:
+        case SWT.TRAVERSE_TAB_NEXT:
+        case SWT.TRAVERSE_TAB_PREVIOUS:
+          if ((programFlagText.getStyle() & SWT.SINGLE) != 0) {
```

Comment:
I have not yet touched/tested this code. First we should see what's the user experience and then see if that's sufficient, issue #83 has been filed to cover it.